src/Control/Distributed/Process/Internal/CQueue.hs

```diff
@@ -101,7 +109,7 @@ dequeue :: forall m a.
         -> BlockSpec         -- ^ Blocking behaviour
         -> [MatchOn m a]     -- ^ List of matches
         -> IO (Maybe a)      -- ^ 'Nothing' only on timeout
-dequeue (CQueue arrived incoming) blockSpec matchons = mask_ $
+dequeue (CQueue arrived incoming size) blockSpec matchons = mask_ $ decrementJust $
```

Comment:
I had a conversation with Edsko about mask usage here, and he said he is fine with that. I can try to find logs if needed.